AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/AWSDataStorePlugin.swift

```diff
@@ -114,58 +114,35 @@ final public class AWSDataStorePlugin: DataStoreCategoryPlugin {
         ModelListDecoderRegistry.registerDecoder(DataStoreListDecoder.self)
     }
 
-    /// Initializes the underlying storage engine
-    /// - Returns: success if the engine is successfully initialized or
-    ///            a failure with a DataStoreError
-    func initStorageEngine() -> InitStorageEngineResult {
-        storageEngineInitQueue.sync {
-            if storageEngine != nil {
-                return .alreadyInitialized
+    func initStorageEngineAsync() -> Promise<StorageEngineBehavior, DataStoreError> {
```

Comment:
Maybe add a `@discardableResult` here since the call sites don't use the promise?

Reply:
The `Promise` is like a continuation builder. It will be used to trigger the runnable closures.